src/sql-server-util/src/cdc.rs

```diff
@@ -223,6 +252,16 @@ impl<'a> CdcStream<'a> {
                         // Ensure LSNs don't go backwards.
                         *instance_lsn = std::cmp::max(*instance_lsn, next_lsn);
                     }
+                } else if curr_lsn >= db_max_lsn {
+                    // We've ingested everything the DB currently has. Emit a progress
+                    // event in case we haven't already.
+                    //
+                    // Note: Generally this only ever occurs when starting a stream and the LSN
+                    // we're starting with matches the current max of the DB.
```

Comment:
I think this else block can be removed entirely and instead just emit a progress event at the beginning before entering the replication loop. Or don't emit it at all and let the caller assume it

Reply:
Good thought, refactored to emit this once at the start!